algorithms/indexing/index.h

```diff
@@ -197,16 +197,17 @@ namespace dials { namespace algorithms {
       // convert into a single array for input to AnnAdaptor
       // based on flex.vec_3.as_double()
       // scitbx/array_family/boost_python/flex_vec3_double.cpp
-      af::shared<double> rlps_double(reciprocal_space_points.size() * 3,
+      af::shared<double> rlps_double(reciprocal_space_points.size() * 4,
                                      af::init_functor_null<double>());
       double* r = rlps_double.begin();
       for (std::size_t i = 0; i < reciprocal_space_points.size(); i++) {
         for (std::size_t j = 0; j < 3; j++) {
           *r++ = reciprocal_space_points[i][j];
         }
+        *r++ = phi[i];
```

Comment:
My only comment here is that stuff like `*r++` always gets me scratching my head as to whether `++` happens before or after the operation etc. - and we already know that this should be written to `rlps_double[4*i+3]` - applies to the line above which is obviously unchanged as well. 

Appreciate that this is consistent with the "house style" locally and does mean an increment vs. add/multiply but I wonder how much difference that makes to the runtime (i.e. compilers are clever & can unroll the add/multiply version vs. cannot unroll the increment version)